lib/IRGen/IRGenModule.cpp

```diff
@@ -927,9 +929,9 @@ void IRGenModule::emitAutolinkInfo() {
                                        }),
                         AutolinkEntries.end());
 
-  if (TargetInfo.OutputObjectFormat == llvm::Triple::COFF ||
-      TargetInfo.OutputObjectFormat == llvm::Triple::MachO ||
-      Triple.isPS4()) {
+  if ((TargetInfo.OutputObjectFormat == llvm::Triple::COFF &&
+       !Triple.isOSCygMing()) ||
```

Comment:
Im familiar with the directive handling in binutils.  That sounds like something else is going wrong.  I know that the directives generated by LLVM should be properly handled by binutils linker.  We should look at the generated object file and figure out what exactly prevents us from using that infrastructure.  If its a bug in LLVM, we can fix that :-).